### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by gsxdsm** [tetra-framework/tetra#81] [General]
@nerdoc @samwillis wanted to discuss the roadmap for the next few months. I know there is lot of support is going into forms. There are a lot of great ideas on the tetra docs site, maybe we can put together a prioritized list of things to tackle if there's not already and I'm happy to work on things in order of property.

**Comment by nerdoc** [comment]
OK, that's a great idea.

 * I've been working on form support the last months, but my time and skills are somehow limited, and I'm mostly stuck at the form upload. I trued a few things, but nothing worked. If you want to help here, that would be the most valuable help, as it would free some resources in my head for other goals in Tetra.
 * The next big thing would be inter-component communication, as mentioned in the discussion about stores,/out of band swaps. Child components should be able to send events or share data with parent components, ideally on both sides (client=Alpine + server=Django)
 * Push notifications using websockets or SSE. Changes on the Server should trigger Client/Alpine events (and maybe hence trigger a reload of a component, this would be the easiest, or at least a data update.

These are already three reply big goals, with the first (upload) being a showstopper for the next point release, as Form support is bad without file upload.

**Reply by gsxdsm** [reply]
Perfect. I'll start looking at form support first and specifically upload. Is there a discussion or notes on what issue you're running into?

**Reply by nerdoc** [reply]
I'll continue here: https://github.com/tetra-framework/tetra/issues/67
